### DISCUSSION TITLE
How cand I Bind an string inside SvgImage Path markup ?

### DISCUSSION THREAD

**Discussion by cesarchefinho** [AvaloniaUI/Avalonia#11812] [Q&A]
How cand I Bind an string inside SvgImage Path markup ?
```xaml
<Image Height="15" Source="{SvgImage {Binding Idioma.Svg} }" />
```
```csharp
public class IdiomaType
{
public string Name { get; set; } = "";
public string Sigla { get; set; } = "";
public string Svg { get; set; } = "";
}

public partial class MainViewModel : ObservableValidator
{
[ObservableProperty]
public IdiomaType idioma = new IdiomaType
{
Name = "Português",
Svg = "/Assets/pt-BR.svg",
Sigla = "pt-BR"
};
}
```
Can someone help and say what am i doing wrong ?

I am geting this error

Unable to convert Avalonia.Markup:Avalonia.Data.Binding to System.Runtime:System.String for constructor of Avalonia.Svg.Skia:Avalonia.Svg.Skia.SvgImageExtension

thanks

**Comment by maxkatz6** [comment]
It seems [SvgImage.Path](https://github.com/wieslawsoltes/Svg.Skia/blob/master/src/Avalonia.Svg/SvgImageExtension.cs#L22) property is not an avalonia property, but just a CLR prop. Meaning you can't bind to it. I wonder if it can be changed cc @wieslawsoltes

As an alternative, you can use IValueConverter or you can try `<SvgImage Height="15" Source="{Binding Idioma.Svg}" />` (haven't tested it).

**Reply by wieslawsoltes** [reply]
It's markup extension so would Avalonia property be even possible to create and work?

**Reply by workgroupengineering** [reply]
if I remember correctly, in Avalonia to define a markup extension it is sufficient that there is a method called 'public object ProvideValue(IServiceProvider serviceProvider)', consequently you could derive your markup extension from AvaloniaObject

**Reply by maxkatz6** [reply]
Yeah, it doesn't need to inherit MarkupExtension

**Comment by michalss** [comment]
does anyone know how to do it pls ?

**Reply by timunie** [reply]
check the samples they provide in Avalonia.SVG pls. https://github.com/wieslawsoltes/Svg.Skia/tree/master/samples

**Reply by michalss** [reply]
well there is nothing about binding the path.. at least i did not find anything...

**Reply by michalss** [reply]
ok if anyone needs it only way how to do it is to use diff lib

```axml
<Svg Name="svgSvg" Height="16" Width="16" Path="{Binding IconPath }" />
```

https://github.com/wieslawsoltes/Svg.Skia/blob/master/samples/AvaloniaSvgSample/MainWindow.axaml
